### ISSUE TITLE
RetryingClientFactory.maxRetries is not documented

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4618]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#4618|
|Type|enhancement|
|Created|2010-07-28 20:30:42Z|
|Branch|https://github.com/twisted/twisted/tree/maxretries-doc-4618-2|

The API docs should explain what it does.

Attachments:

* [4618-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e12/e127a19374d8ae8182532742050f04e2dd3e4072/4adef36513820a2897f7d2451f65eb137ac01eea.patch) (686 bytes) - added by awclin on 2011-01-22 13:07:41Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4618 4618
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_maxretries_doc_4618_2 branches/maxretries-doc-4618-2
branch_author__thijs__exarkun__awclin thijs, exarkun, awclin
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy_documentation easy documentation
time__1280349042000000 1280349042000000
changetime__1300028944000000 1300028944000000
version__None None
owner__ 
cc__awclin cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="awclin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|awclin commented|
|-|-|

I hope that a) this is as simple as it seems, b) I am correct in assuming you meant ReconnectingClientFactory, and c) my description/use of markup is adequate.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [30556]) Branching to 'maxretries-doc-4618'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [30557]) Apply 4618-1.patch, refs [#4618](https://github.com/twisted/twisted/issues/4618)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [30558]) Add type for vars and news file, refs [#4618](https://github.com/twisted/twisted/issues/4618)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  1. none of `maxDelay`, `initialDelay`, `factor`, `jitter` are restricted to being ints.  The default values for three of them are even floats.
  1. `clock` is also not an int, it's an `IReactorTime` provider.
  1. The news file doesn't use the style recommended on ReviewProcess#Newsfile: _Generally, the grammatical subject of the sentence should be a Python object, and the verb should be something that the object does, prefixed with "now"._  For example, _The maxRetries attribute of twisted.internet.protocols.RetryingClientFactory now has API documentation._

The `maxRetries` docs look good though. :)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [30593]) remove incorrect type docs, fix up news file. refs [#4618](https://github.com/twisted/twisted/issues/4618)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_6):
>   1. none of `maxDelay`, `initialDelay`, `factor`, `jitter` are restricted to being ints.  The default values for three of them are even floats.

I removed those type docs.

>   1. `clock` is also not an int, it's an `IReactorTime` provider.

Changed type to `IReactorTime`.

>   1. The news file doesn't use the style recommended on ReviewProcess#Newsfile: _Generally, the grammatical subject of the sentence should be a Python object, and the verb should be something that the object does, prefixed with "now"._  For example, _The maxRetries attribute of twisted.internet.protocols.RetryingClientFactory now has API documentation._

Thanks, fixed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Thanks!  Looks good.  Please merge.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [30981]) Branching to 'maxretries-doc-4618-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [30983]) Merge maxretries-doc-4618-2 

Author: awclin, thijs
Reviewer: exarkun
Fixes: [#4618](https://github.com/twisted/twisted/issues/4618)

Add improved API documentation for `ReconnectingClientFactory`, in particular
the `maxRetries` attribute.
